hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -797,14 +831,14 @@ private int addToReport(String jobId, int index,
   private synchronized void updateJobStatus(String jobKey,
                                             JobStatus oldStatus,
                                             JobStatus newStatus) {
-    SnapshotDiffJob report = snapDiffJobTable.get(jobKey);
-    if (report.getStatus() != oldStatus) {
+    SnapshotDiffJob snapshotDiffJob = snapDiffJobTable.get(jobKey);
+    if (snapshotDiffJob.getStatus() != oldStatus) {
       throw new IllegalStateException("Invalid job status. Current job " +
-          "status is '" + report.getStatus() + "', while '" + oldStatus +
-          "' is expected.");
+          "status is '" + snapshotDiffJob.getStatus() + "', while '" +
+          oldStatus + "' is expected.");
     }
-    snapDiffJobTable.put(jobKey,
-        new SnapshotDiffJob(report.getJobId(), newStatus));
+    snapshotDiffJob.setStatus(newStatus);
+    snapDiffJobTable.put(jobKey, snapshotDiffJob);
   }
```

Comment:
gotta make sure this write goes through Ratis later. Otherwise follower OMs won't have this entry at all (thus no "resume" when former leader is permanantly down).

Reply:
Now that we decided to intentionally prevent SnapDiff job status from going through Ratis. Let's keep the SnapDiff job status in the separate DB and not in the main OM DB.

Reply:
As of now, all the snapDiff related tables are in separate DB: [db.snapdiff](https://github.com/apache/ozone/blob/5756fd876f0997629205238195edfee0c7472b01/hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConsts.java#L567).